core/ledger/ledger_test.go

```diff
@@ -902,3 +902,24 @@ func TestCopyState(t *testing.T) {
 	values, _ := l.GetStateMultipleKeys("chaincodeID2", []string{"key1", "key2", "key3"}, true)
 	testutil.AssertEquals(t, values, [][]byte{[]byte("value1"), []byte("value2"), []byte("value3")})
 }
+
+func TestLedgerEmptyArrayValue(t *testing.T) {
```

Comment:
Could you add tests to confirm the error is returned on a empty key or a nil value?